platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/Bucket.java

```diff
@@ -82,23 +83,47 @@
      * entries, and entry data. Buffer is expanded as needed, when new entries are added. Buffer
      * limit is kept equal to the bucket size in bytes.
      */
-    private ByteBuffer bucketBuffer;
+    private volatile ByteBuffer bucketBuffer;
```

Comment:
I am slightly worried about this buffer. The field is changed to volatile, as it's accessed from multiple threads (not in parallel, though). However, buffer content is not synchronized explicitly